src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs

```diff
@@ -58,14 +58,17 @@ public void Dispose()
             var numberOfChecksumsToSearch = checksumsToFind.Count;
             Contract.ThrowIfTrue(checksumsToFind.Contains(Checksum.Null));
 
-            await FindAssetsAsync(assetPath, checksumsToFind, assetMap, cancellationToken).ConfigureAwait(false);
+            await FindAssetsAsync(
+                assetPath, checksumsToFind,
+                (checksum, asset) => assetMap[checksum] = asset,
```

Comment:
> (checksum, asset) => assetMap[checksum] = asset, [](http://example.com/codeflow?start=15&length=49)

Containing PR gets rid of the closure allocation (or will once [https://github.com/dotnet/roslyn/pull/73013#discussion_r1564749223](https://github.com/dotnet/roslyn/pull/73013#discussion_r1564749223) is addressed)